File: repository/src/main/java/com/SReMake/repository/user/UserRepository.java
package com.SReMake.repository.user;


import com.SReMake.model.user.User;
import org.babyfish.jimmer.spring.repo.support.AbstractJavaRepository;
import org.babyfish.jimmer.sql.JSqlClient;
import org.springframework.stereotype.Repository;

@Repository
public class UserRepository extends AbstractJavaRepository<User, Long> {
    public UserRepository(JSqlClient sql) {
        super(sql);
    }
}

File: repository/src/main/kotlin/com/SReMake/repository/user/UserRepository.kt
package com.SReMake.repository.user

import com.SReMake.model.user.User
import com.SReMake.model.user.dto.UserSearchInput
import org.babyfish.jimmer.Page
import org.babyfish.jimmer.spring.repo.KotlinRepository
import org.babyfish.jimmer.spring.repo.PageParam


interface UserRepository : KotlinRepository<User, Long> {
    /**
     * 分页查询
     */
    fun findPage(pageParam: PageParam, params: UserSearchInput?): Page<User>

    fun findByUsername(username: String): User?
}